src/gpu/intel/jit/gemm/include/type.hpp

```diff
@@ -99,13 +99,13 @@ class Type {
     }
     constexpr Type baseType() const { return *this; }
 
-    template <typename U> constexpr friend int operator*(U a, Type t) {
-        return t.isInt4() ? int((unsigned(a) + 1) >> 1) : int(a * (U(1) << t.log2Size()));
+    template <typename U> constexpr friend U operator*(U a, Type t) {
+        return t.isInt4() ? (a + 1) >> 1 : a * (U(1) << t.log2Size());
```

Comment:
Coverity will complain about applying right shifts to signed types...
```suggestion
        return t.isInt4() ? (uint64_t(a) + 1) >> 1 : a * (U(1) << t.log2Size());
```

Reply:
I just changed this to `a+1/2`, the optimization to using shifts can easily be done by the compiler.

Reply:
This isn't the correct behavior for negative `a`, unfortunately.

Reply:
Fixed, and I added a couple static assertions to ensure this is maintained.